### ISSUE TITLE
Change container of opus audio

### ISSUE THREAD

**Issue by victordesfe** [ytdl-org/youtube-dl#22490]
## Checklist

- [x ] I'm reporting a feature request
- [ x] I've verified that I'm running youtube-dl version **2019.09.12.1**
- [x ] I've searched the bugtracker for similar feature requests including closed ones

## Description

Hi, with yt-dl i can download the audio from any video, which is great. The thing is, youtube stores opus audio in the webm container.

Is there any function to change the container within the yt-dl process? I tried:
`youtube-dl.exe -f "bestaudio[ext=webm]" --batch-file "youtube-dl.txt" --merge-output-format ogg -v`
but does nothing, it'll still give you the webm container file.
I believe it's not implemented or something, is there any known workaround?
thanks

**Comment by victordesfe** [comment]
also wanted to note, adding .mka (matroska audio container) support would be great, but the main one i'm looking for is the ogg container with .opus extension, as it's the recommended by the original opus specification.

**Comment by victordesfe** [comment]
I've found a way to do this with ffmpeg:
`ffmpeg -i input.webm -vn -acodec copy out.opus`
however i still think it would be great to get it integrated with yt-dl

**Comment by nao20010128nao** [comment]
You might want
```
youtube-dl --audio-format=opus -x
```
It gives a audio file with .opus extension.

**Comment by mcepl** [comment]
> You might want
> 
> ```
> youtube-dl --audio-format=opus -x
> ```
> 
> It gives an audio file with .opus extension.

That's helpful, but not what the original issue was about. ``--merge-output-format ogg`` should create file in the ogg container, which is widely more recognized than ``.opus`` files. For example, many Android phones can play Opus-encoded files, but they don’t recognize ``.opus`` files, only ``.ogg`` ones.

**Comment by mcepl** [comment]
What’s actually interesting is that apparently the files **IS** in Ogg container, just the extension is wrong. See
```
~@stitny$ ffplay -stats Dick\ Walters\ -\ Hot\ Liquorice-KNMoYJhicEY.opus 
ffplay version 4.2.1 Copyright (c) 2003-2019 the FFmpeg developers
  built with gcc 9 (SUSE Linux)
[...]
Input #0, ogg, from 'Dick Walters - Hot Liquorice-KNMoYJhicEY.opus':
  Duration: 00:02:52.45, start: 0.007500, bitrate: 138 kb/s
    Stream #0:0(eng): Audio: opus, 48000 Hz, stereo, fltp
    Metadata:
      encoder         : Lavf58.29.100
  16.14 M-A: -0.000 fd=   0 aq=   22KB vq=    0KB sq=    0B f=0/0   
~@stitny$
```

**Comment by bitraid** [comment]
The extension is not wrong because an `.opus` file is actually an ogg container with opus audio (opus is a codec and not a conainer).

**Comment by mcepl** [comment]
> The extension is not wrong because an `.opus` file is actually an ogg container with opus audio (opus is a codec and not a conainer).

Exactly, so if ``youtube-dl`` named the generated file ``.ogg`` this ticket would be resolved.

**Comment by bitraid** [comment]
This ticket is about the container and not the extension.

**Comment by mcepl** [comment]
> This ticket is about the container and not the extension.

My point was that this ticket is NOTABUG with regards to the container. Output is already in OGG.

**Comment by icheyne** [comment]
I have exactly this problem too. I can download an opus file but I can't play it in Android without renaming it manually to file.ogg. Would be good to know how to get youtube-dl to rename it automatically or just convert it to ogg with FFMPEG.

**Comment by mcepl** [comment]
> Would be good to know how to get youtube-dl to rename it automatically or just convert it to ogg with FFMPEG.

You don’t have to use ffmpeg. ``mv filename.opus filename.ogg`` is enough.

**Comment by icheyne** [comment]
Sorry I should have explained that I'm using **youtube-dl.conf** with the following config:

```
-x
-f worstaudio[ext=m4a]
-o "C:\Users\iain\Sync\audio\%(title)s.%(ext)s"
```

I'm using m4a because opus files won't play in Android. Woudl be awesome to just use -`f worstaudio` and have a playable file.

**Comment by mcepl** [comment]
> I'm using m4a because opus files won't play in Android.

That is not correct. Opus files (renamed to ``.ogg``) do play with the native Android media player.

**Comment by pihug12** [comment]
It's not clean, but `--exec 'ffmpeg -i {} -vn -acodec copy "$(basename -- {} .webm).ogg" && rm {}'` seems to be working with Opus format as input.

Example: `youtube-dl -f 250 --exec 'ffmpeg -i {} -vn -acodec copy "$(basename -- {} .webm).ogg" && rm {}' https://www.youtube.com/watch?v=2EaxYi31ips`

**Comment by icheyne** [comment]
That's a decent solution. Thank  you.

**Comment by tryamid** [comment]
Extension lies but magic code doesn't.

**Comment by mixmastamyk** [comment]
`-x, --extract-audio`

When I use this with format 251 (best opus), the option grabs the audio track from the webm and puts it in an ogg container with an opus extension.  I have a wrapper script that does it automatically for me.

**Comment by tripulse** [comment]
That is *service dependent*, `251` code applies for YouTube Opus@WebM streams and **not for** anything else.
